ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/BasicActionExpiryScannerTest.java

```diff
@@ -0,0 +1,129 @@
+import com.arjuna.ats.arjuna.objectstore.StateStatus;
+import com.arjuna.ats.arjuna.objectstore.StoreManager;
+import com.arjuna.ats.arjuna.state.OutputObjectState;
+import com.arjuna.ats.internal.arjuna.common.UidHelper;
+import com.arjuna.ats.internal.arjuna.recovery.AdvancedAtomicActionExpiryScanner;
+import com.arjuna.ats.internal.arjuna.recovery.AdvancedAtomicActionPurgeExpiryScanner;
+
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * This test class covers multiple {@code ExpiryScanner} classes.
+ *
+ * @see BasicActionExpiryScanner
+ * @see AdvancedAtomicActionExpiryScanner
+ * @see AdvancedAtomicActionPurgeExpiryScanner
+ */
+public class BasicActionExpiryScannerTest {
+
+    private static final Logger log = Logger.getLogger(BasicActionExpiryScannerTest.class);
+    private static final String TYPE_NAME = new AtomicAction().type();
+    private static final RecoveryStore recoveryStore = StoreManager.getRecoveryStore();
+
+    @BeforeAll
+    public static void beforeAll() {
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setExpiryScanInterval(1);
+    }
+
+    public OutputObjectState createObjectState(Uid txId, Uid processUid) throws IOException {
+        OutputObjectState os = new OutputObjectState();
+        os.packStringBytes("#ARJUNA#".getBytes(StandardCharsets.UTF_8));
+        UidHelper.packInto(txId, os);
+        UidHelper.packInto(processUid, os);
+        os.packLong(Instant.now().minus(Duration.ofHours(1)).toEpochMilli());
+        os.packBoolean(true);
+        os.packInt(RecordType.NONE_RECORD);
+        os.packInt(0);
+        os.packInt(ActionStatus.H_HAZARD);
+        os.packInt(ActionType.TOP_LEVEL);
+        os.packInt(TwoPhaseOutcome.HEURISTIC_HAZARD);
+        return os;
+    }
+
+    @Test
+    public void testAdvancedAtomicActionExpiryScanner() {
+        boolean passed = false;
+        Uid txId = new Uid();
+        Uid processUid = new Uid();
+
+        try {
+            OutputObjectState os = createObjectState(txId, processUid);
+
+            log.debug("Creating dummy log");
+            recoveryStore.write_committed(txId, TYPE_NAME, os);
+            if (recoveryStore.currentState(txId, TYPE_NAME) == StateStatus.OS_COMMITTED) {
+                log.debugf("Wrote dummy transaction %n", txId);
+            }
+
+            AdvancedAtomicActionExpiryScanner scanner = new AdvancedAtomicActionExpiryScanner();
+            scanner.scan();
+
+            if (recoveryStore.currentState(txId, TYPE_NAME + "/Expired") == StateStatus.OS_COMMITTED) {
+                log.debug("Transaction log moved!");
+                passed = true;
+            } else {
+                log.debug("Transaction log not moved!");
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        assertTrue(passed);
```

Comment:
My IDE highlights this because you are importing both junit 4 and junit 5 and it suggests replacing it with `Assertions.assertTrue(passed);`